flow/visualize/transfer/i210_transfer.py

```diff
@@ -0,0 +1,372 @@
+"""Transfer and replay for i210 environment."""
+import argparse
+from copy import deepcopy
+import numpy as np
+import os
+import time
+
+import ray
+try:
+    from ray.rllib.agents.agent import get_agent_class
+except ImportError:
+    from ray.rllib.agents.registry import get_agent_class
+from ray.tune.registry import register_env
+
+from flow.core.util import emission_to_csv
+from flow.utils.registry import make_create_env
+from flow.utils.rllib import get_flow_params
+from flow.utils.rllib import get_rllib_config
+from flow.utils.rllib import get_rllib_pkl
+
+from flow.visualize.transfer.util import inflows_range
+
+from examples.exp_configs.rl.multiagent.multiagent_i210 import flow_params as I210_MA_DEFAULT_FLOW_PARAMS
+
+EXAMPLE_USAGE = """
+example usage:
+    python ./visualizer_rllib.py /ray_results/experiment_dir/result_dir 1
+
+Here the arguments are:
+1 - the path to the simulation results
+2 - the number of the checkpoint
+"""
+
+
+def run_transfer_test(args, flow_params, transfer_test, rllib_config=None, result_dir=None):
+    """Run transfer test (defined by transfer_fn) by modifying flow_params.
+
+    Arguments:
+    ---------
+        args {[Namespace]} -- [args from argparser]
+        flow_params {[flow_params object, pulled from ]} -- [description]
+        transfer_fn {[type]} -- [description]
+
+    Keyword Arguments:
+    -----------------
+        rllib_config {[type]} -- [description] (default: {None})
+        result_dir {[type]} -- [description] (default: {None})
+    """
+    flow_params = transfer_test.flow_params_modifier_fn(flow_params)
+
+    sim_params = flow_params['sim']
+    sim_params.num_clients = 1
+
+    sim_params.restart_instance = True
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    emission_path = '{0}/test_time_rollout/'.format(dir_path)
+    sim_params.emission_path = emission_path if args.gen_emission else None
+
+    # pick your rendering mode
+    if args.render_mode == 'sumo_web3d':
+        sim_params.num_clients = 2
+        sim_params.render = False
+    elif args.render_mode == 'drgb':
+        sim_params.render = 'drgb'
+        sim_params.pxpm = 4
+    elif args.render_mode == 'sumo_gui':
+        sim_params.render = False  # will be set to True below
+    elif args.render_mode == 'no_render':
+        sim_params.render = False
+    if args.save_render:
+        if args.render_mode != 'sumo_gui':
+            sim_params.render = 'drgb'
+            sim_params.pxpm = 4
+        sim_params.save_render = True
+
```

Comment:
Feel like this code is duplicated in a lot of places. Can we move it all into one spot?